### PR TITLE
- Refactor ember-debug to use ES6.

### PR DESCRIPTION
**Pull request by Mawaheb**
OK, I believe this is ready, But, there are a few problems that I faced:

- I couldn't refactor the contents of `/vendor`, when I tried, I got an error "Can't use block-scoped let, const out of restricted mode."

- Because arrow functions should always be anonymous, I couldn't/figure out how to refactor [this one](https://github.com/emberjs/ember-inspector/blob/master/ember_debug/view-debug.js#L192) to use `this` instead of `self`
similar problem with [this one](https://github.com/emberjs/ember-inspector/blob/master/ember_debug/deprecation-debug.js#L177)

As always, Any suggestions, comments, notes, things to be changed or feedback is always appreciated

### CODE REVIEW

**Comment by teddyzeenny** [comment]
:+1: Thanks @Mawaheb! I will review this weekend at the latest.

**Comment by Mawaheb** [comment]
No problem :) My pleasure.

File: ember_debug/adapters/basic.js [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
-var Promise = RSVP.Promise;
-
-export default Ember.Object.extend({
-  init: function() {
-    var self = this;
-    RSVP.resolve(this.connect(), 'ember-inspector').then(function() {
-      self.onConnectionReady();
+const Ember = window.Ember;
+const { $, A, computed, RSVP, Object } = Ember;
+const { Promise } = RSVP;
```
Comment (Patsy-issa) [inline ember_debug/adapters/basic.js]: `const { Promise, resolve } = RSVP;`

File: ember_debug/adapters/basic.js [hunk 2]
```diff
@@ -1,15 +1,12 @@
...
+
+export default Object.extend({
+  init() {
+    RSVP.resolve(this.connect(), 'ember-inspector').then(() => {
```
Comment (Patsy-issa) [inline ember_debug/adapters/basic.js]: `resolve(...`

File: ember_debug/adapters/basic.js [hunk 3]
```diff
@@ -69,11 +66,11 @@ export default Ember.Object.extend({
...
+  inspectElement(/* elem */) {},
 
-  _messageCallbacks: Ember.computed(function() { return Ember.A(); }).property(),
+  _messageCallbacks: computed(function() { return A(); }),
 
-  _messageReceived: function(message) {
+  _messageReceived(message) {
     this.get('_messageCallbacks').forEach(function(callback) {
       callback.call(null, message);
     });
```
Comment (Patsy-issa) [inline ember_debug/adapters/basic.js]:
```javascript
this.get('_messageCallbacks').forEach(callback => callback.call(null, message));
```

File: ember_debug/adapters/basic.js [hunk 4]
```diff
@@ -149,12 +145,11 @@ export default Ember.Object.extend({
...
+  onConnectionReady() {
     // Flush pending messages
-    var self = this;
-    var messages = this.get('_pendingMessages');
-    messages.forEach(function(options) {
-      self.sendMessage(options);
+    const messages = this.get('_pendingMessages');
+    messages.forEach((options) => {
+      this.sendMessage(options);
     });
```
Comment (Patsy-issa) [inline ember_debug/adapters/basic.js]:
```javascript
messages.forEach(options => this.sendMessage(options));
```

**Comment by Mawaheb** [comment]
Thanks for your feedback @Patsy-issa  :) .

**Comment by Patsy-issa** [comment]
Np ^^ I ll take a look at the other files when I have the time.

File: ember_debug/adapters/basic.js [hunk 5]
```diff
@@ -1,15 +1,12 @@
...
-var RSVP = Ember.RSVP;
-var Promise = RSVP.Promise;
-
-export default Ember.Object.extend({
-  init: function() {
-    var self = this;
-    RSVP.resolve(this.connect(), 'ember-inspector').then(function() {
-      self.onConnectionReady();
+const Ember = window.Ember;
+const { $, A, computed, RSVP, Object } = Ember;
```
Comment (teddyzeenny) [inline ember_debug/adapters/basic.js]: `Object` has meaning in native javascript. You should rename it to something like `EmberObject`:
`const { $, A, computed, RSVP, Object: EmberObject } = Ember;`

File: ember_debug/adapters/chrome.js [hunk 6]
```diff
@@ -1,42 +1,41 @@
...
 
-var ChromeAdapter = BasicAdapter.extend({
-  connect: function() {
-    var channel = this.get('_channel');
-    var self = this;
-    return this._super.apply(this, arguments).then(function() {
+const ChromeAdapter = BasicAdapter.extend({
+  connect() {
+    const channel = this.get('_channel');
+    return this._super.apply(this, arguments).then(() => {
```
Comment (teddyzeenny) [inline ember_debug/adapters/chrome.js]: ES6 spread operator: `this._super(...arguments)`

File: ember_debug/adapters/firefox.js [hunk 7]
```diff
@@ -1,14 +1,15 @@
 /* eslint no-empty:0 */
 import BasicAdapter from "./basic";
-var Ember = window.Ember;
+const Ember = window.Ember;
+const { run } = Ember;
 
-var FirefoxAdapter = BasicAdapter.extend({
-  init: function() {
+const FirefoxAdapter = BasicAdapter.extend({
```
Comment (teddyzeenny) [inline ember_debug/adapters/firefox.js]: Let's just export default instead of two steps.

File: ember_debug/adapters/websocket.js [hunk 8]
```diff
@@ -1,49 +1,45 @@
 import BasicAdapter from "./basic";
-var Ember = window.Ember;
-var computed = Ember.computed;
-var Promise = Ember.RSVP.Promise;
-var $ = Ember.$;
+const Ember = window.Ember;
+const { $, computed, run, RSVP: { Promise } } = Ember;
 
-var WebsocketAdapter = BasicAdapter.extend({
+const WebsocketAdapter = BasicAdapter.extend({
```
Comment (teddyzeenny) [inline ember_debug/adapters/websocket.js]: `export default BasicAdapter.extend`

File: ember_debug/adapters/websocket.js [hunk 9]
```diff
@@ -1,49 +1,45 @@
...
 
-  _disconnect: function() {
+  _disconnect() {
     this.get('socket').removeAllListeners("emberInspectorMessage");
   },
 
-  connect: function() {
-    var self = this;
+  connect() {
     return new Promise(function(resolve, reject) {
```
Comment (teddyzeenny) [inline ember_debug/adapters/websocket.js]: `return new Promise((resolve, reject) => {`

File: ember_debug/addons/ember-new-computed/index.js [hunk 10]
```diff
@@ -1,21 +1,21 @@
...
-var Ember = window.Ember;
-var computed = Ember.computed;
+const Ember = window.Ember;
+const { computed } = Ember;
 
 export default function() {
-  var polyfillArguments = [];
-  var config = arguments[arguments.length - 1];
+  let polyfillArguments = [],
+      config = arguments[arguments.length - 1];
```
Comment (teddyzeenny) [inline ember_debug/addons/ember-new-computed/index.js]: Single variable declaration with assignment (same as Ember core https://github.com/emberjs/ember.js/pull/10037).
```js
let polyfillArguments = [];
let config = arguments[arguments.length - 1];
```

File: ember_debug/container-debug.js [hunk 11]
```diff
@@ -1,19 +1,18 @@
 import PortMixin from "ember-debug/mixins/port-mixin";
-var Ember = window.Ember;
-var EmberObject = Ember.Object;
-var computed = Ember.computed;
-var oneWay = computed.oneWay;
+const Ember = window.Ember;
+const { Object, computed } = Ember;
```
Comment (teddyzeenny) [inline ember_debug/container-debug.js]: `const { Object: EmberObject, computed } = Ember;`

File: ember_debug/container-debug.js [hunk 12]
```diff
@@ -24,29 +23,29 @@ export default EmberObject.extend(PortMixin, {
...
+  shouldHide(type) {
     return type[0] === '-' || this.get('TYPES_TO_SKIP').indexOf(type) !== -1;
   },
 
-  instancesByType: function() {
-    var key, instancesByType = {};
-    var cache = this.get('container').cache;
+  instancesByType() {
+    let key, instancesByType = {},
+        cache = this.get('container').cache;
```
Comment (teddyzeenny) [inline ember_debug/container-debug.js]:
```js
let key;
let instancesByType = {};
let cache = this.get('container').cache;
```

File: ember_debug/container-debug.js [hunk 13]
```diff
@@ -59,37 +58,37 @@ export default EmberObject.extend(PortMixin, {
     return instancesByType;
   },
 
-  getTypes: function() {
-    var key, types = [];
-    var instancesByType = this.instancesByType();
+  getTypes() {
+    let key, types = [];
```
Comment (teddyzeenny) [inline ember_debug/container-debug.js]:
```js
let key;
let types = [];
```

File: ember_debug/data-debug.js [hunk 14]
```diff
@@ -1,7 +1,10 @@
 import PortMixin from 'ember-debug/mixins/port-mixin';
-var Ember = window.Ember;
-var DataDebug = Ember.Object.extend(PortMixin, {
-  init: function() {
+const Ember = window.Ember;
+const { Object, computed, guidFor } = Ember;
```
Comment (teddyzeenny) [inline ember_debug/data-debug.js]: `Object: EmberObject`

File: ember_debug/data-debug.js [hunk 15]
```diff
@@ -1,7 +1,10 @@
...
+const { alias } = computed;
+
+const DataDebug = Object.extend(PortMixin, {
```
Comment (teddyzeenny) [inline ember_debug/data-debug.js]: `export default EmberObject.extend`

File: ember_debug/data-debug.js [hunk 16]
```diff
@@ -36,34 +39,33 @@ var DataDebug = Ember.Object.extend(PortMixin, {
...
 
-  modelTypesAdded: function(types) {
-    var self = this, typesToSend;
-    typesToSend = types.map(function(type) {
-      return self.wrapType(type);
+  modelTypesAdded(types) {
+    let typesToSend;
+    typesToSend = types.map((type) => {
+      return this.wrapType(type);
     });
```
Comment (teddyzeenny) [inline ember_debug/data-debug.js]: Tip: single-line arrow functions can have implicit returns if you don't use curly brackets:
```js
types.map(type => this.wrapType(type));
```

File: ember_debug/data-debug.js [hunk 17]
```diff
@@ -36,34 +39,33 @@ var DataDebug = Ember.Object.extend(PortMixin, {
...
   },
 
-  modelTypesUpdated: function(types) {
-    var self = this;
-    var typesToSend = types.map(function(type) {
-      return self.wrapType(type);
+  modelTypesUpdated(types) {
+    let typesToSend = types.map((type) => {
+      return this.wrapType(type);
     });
```
Comment (teddyzeenny) [inline ember_debug/data-debug.js]: `let typesToSend = types.map(type => this.wrapType(type));`

File: ember_debug/data-debug.js [hunk 18]
```diff
@@ -75,40 +77,39 @@ var DataDebug = Ember.Object.extend(PortMixin, {
...
   },
 
-  wrapRecord: function(record) {
-    var objectId = Ember.guidFor(record.object);
-    var columnValues = {};
-    var searchKeywords = [];
+  wrapRecord(record) {
+    const objectId = guidFor(record.object);
+    let columnValues = {},
+        searchKeywords = [];
```
Comment (teddyzeenny) [inline ember_debug/data-debug.js]:
```js
let columnValues = {};
let searchKeywords = [];
```

File: ember_debug/data-debug.js [hunk 19]
```diff
@@ -140,67 +141,57 @@ var DataDebug = Ember.Object.extend(PortMixin, {
...
         },
-        function(recordsUpdated) {
-          self.recordsUpdated(recordsUpdated);
+        (recordsUpdated) => {
+          this.recordsUpdated(recordsUpdated);
         },
-        function() {
-          self.recordsRemoved.apply(self, arguments);
+        () => {
+          this.recordsRemoved.apply(this, arguments);
```
Comment (teddyzeenny) [inline ember_debug/data-debug.js]: `this.recordsRemoved(...arguments);`

File: ember_debug/deprecation-debug.js [hunk 20]
```diff
@@ -1,14 +1,11 @@
...
-var EmberObject = Ember.Object;
-var computed = Ember.computed;
-var oneWay = computed.oneWay;
-var run = Ember.run;
-var guidFor = Ember.guidFor;
-var RSVP = Ember.RSVP;
-
-export default EmberObject.extend(PortMixin, {
+const Ember = window.Ember;
+const { Object, computed, guidFor, run, RSVP, A } = Ember;
```
Comment (teddyzeenny) [inline ember_debug/deprecation-debug.js]: `Object: EmberObject`

File: ember_debug/deprecation-debug.js [hunk 21]
```diff
@@ -59,21 +55,20 @@ export default EmberObject.extend(PortMixin, {
...
-    var promises = RSVP.all(this.get('deprecationsToSend').map(function(deprecation) {
-      var obj;
-      var promise = RSVP.resolve(undefined, 'ember-inspector');
-      self.get('deprecations').pushObject(deprecation);
-      var grouped = self.get('groupedDeprecations');
-      var id = guidFor(deprecation.message);
+  sendPending() {
+    let deprecations = A();
+
+    let promises = RSVP.all(this.get('deprecationsToSend').map((deprecation) => {
```
Comment (teddyzeenny) [inline ember_debug/deprecation-debug.js]: `let promises = all(`

File: ember_debug/deprecation-debug.js [hunk 22]
```diff
@@ -59,21 +55,20 @@ export default EmberObject.extend(PortMixin, {
...
+      let obj,
+          promise = resolve(undefined, 'ember-inspector'),
+          grouped = this.get('groupedDeprecations');
```
Comment (teddyzeenny) [inline ember_debug/deprecation-debug.js]: Single variable declaration with assignment.

File: ember_debug/deprecation-debug.js [hunk 23]
```diff
@@ -98,34 +93,34 @@ export default EmberObject.extend(PortMixin, {
...
         deprecations.addObject(obj);
       });
-    }, this));
+    }));
 
-    promises.then(function() {
-      self.sendMessage('deprecationsAdded', {
+    promises.then(() => {
+      this.sendMessage('deprecationsAdded', {
         deprecations: deprecations
```
Comment (teddyzeenny) [inline ember_debug/deprecation-debug.js]: `this.sendMessage('deprecationsAdded', { deprecations });`

File: ember_debug/deprecation-debug.js [hunk 24]
```diff
@@ -137,49 +132,49 @@ export default EmberObject.extend(PortMixin, {
...
     this.originalDeprecate = Ember.deprecate;
 
     Ember.deprecate = function(message, test, options) {
       /* global __fail__*/
       // Code taken from https://github.com/emberjs/ember.js/blob/master/packages/ember-debug/lib/main.js
-      var noDeprecation;
+      let noDeprecation;
 
-      if (typeof test === 'function' && !(Ember.Object.detect(test))) {
+      if (typeof test === 'function' && !(Object.detect(test))) {
```
Comment (teddyzeenny) [inline ember_debug/deprecation-debug.js]: `if (typeof test === 'function' && !(EmberObject.detect(test))) {`

File: ember_debug/general-debug.js [hunk 25]
```diff
@@ -1,10 +1,10 @@
 /* eslint no-empty:0 */
 import PortMixin from "ember-debug/mixins/port-mixin";
-var Ember = window.Ember;
-var computed = Ember.computed;
-var oneWay = computed.oneWay;
+const Ember = window.Ember;
+const { $, computed, Object, A } = Ember;
```
Comment (teddyzeenny) [inline ember_debug/general-debug.js]: `Object: EmberObject`

File: ember_debug/general-debug.js [hunk 26]
```diff
@@ -31,27 +30,27 @@ var GeneralDebug = Ember.Object.extend(PortMixin, {
...
 
   messages: {
-    applicationBooted: function() {
+    applicationBooted() {
       this.sendBooted();
     },
-    getLibraries: function() {
-      var libraries = Ember.libraries;
+    getLibraries() {
+      let libraries = Ember.libraries;
```
Comment (teddyzeenny) [inline ember_debug/general-debug.js]: Is it possible to import `libraries` as a const at the top of the file?
`const { libraries } = Ember`

File: ember_debug/general-debug.js [hunk 27]
```diff
@@ -62,14 +61,14 @@ var GeneralDebug = Ember.Object.extend(PortMixin, {
...
-    refresh: function() {
+    refresh() {
       window.location.reload();
     }
   }
 });
 
 function arrayize(enumerable) {
-  return Ember.A(enumerable).map(function(item) {
+  return A(enumerable).map(function(item) {
```
Comment (teddyzeenny) [inline ember_debug/general-debug.js]: `return A(enumerable).map(item => item);`

File: ember_debug/libs/promise-assembler.js [hunk 28]
```diff
@@ -6,34 +6,34 @@
  */
 
 import Promise from 'ember-debug/models/promise';
-var Ember = window.Ember;
+const Ember = window.Ember;
+const { Object, Evented, A, computed, RSVP, copy, isNone } = Ember;
```
Comment (teddyzeenny) [inline ember_debug/libs/promise-assembler.js]: `Object: EmberObject`

File: ember_debug/libs/promise-assembler.js [hunk 29]
```diff
@@ -6,34 +6,34 @@
...
 
-var PromiseAssembler = Ember.Object.extend(Ember.Evented, {
+let PromiseAssembler = Object.extend(Evented, {
   // RSVP lib to debug
-  RSVP: Ember.RSVP,
+  RSVP: RSVP,
```
Comment (teddyzeenny) [inline ember_debug/libs/promise-assembler.js]: You can just use: 
```js
// RSVP lib to debug
RSVP,
```

File: ember_debug/libs/source-map.js [hunk 30]
```diff
@@ -3,15 +3,12 @@
  * A lot of the code is inspired by/taken from
  * https://github.com/evanw/node-source-map-support
  */
-var Ember = window.Ember;
-var EmberObject = Ember.Object;
-var computed = Ember.computed;
-var RSVP = Ember.RSVP;
-var resolve = RSVP.resolve;
+const Ember = window.Ember;
+const { Object, A, computed, RSVP: { resolve, Promise } } = Ember;
```
Comment (teddyzeenny) [inline ember_debug/libs/source-map.js]: `Object: EmberObject`

File: ember_debug/libs/source-map.js [hunk 31]
```diff
@@ -24,17 +21,16 @@ export default EmberObject.extend({
...
-    var array = Ember.A();
-    var lastPromise = null;
-    parsed.forEach(function(item) {
-      lastPromise = self.get('_lastPromise').then(function() {
-        return self.getSourceMap(item.url);
-      }, null, 'ember-inspector').then(function(smc) {
+  map(stack) {
+    const parsed = A(fromStackProperty(stack));
+    let array = A(),
+        lastPromise = null;
```
Comment (teddyzeenny) [inline ember_debug/libs/source-map.js]:
```js
let array = A();
let lastPromise = null;
```

File: ember_debug/libs/source-map.js [hunk 32]
```diff
@@ -57,13 +53,13 @@ export default EmberObject.extend({
     return {};
   }),
 
-  getSourceMap: function(url) {
-    var sourceMaps = this.get('sourceMapCache');
+  getSourceMap(url) {
+    let sourceMaps = this.get('sourceMapCache');
     if (sourceMaps[url] !== undefined) { return resolve(sourceMaps[url], 'ember-inspector'); }
     return retrieveSourceMap(url).then(function(response) {
```
Comment (teddyzeenny) [inline ember_debug/libs/source-map.js]: `.then(response => {`

File: ember_debug/main.js [hunk 33]
```diff
@@ -11,9 +11,10 @@ import ContainerDebug from "ember-debug/container-debug";
 import DeprecationDebug from "ember-debug/deprecation-debug";
 import Session from "ember-debug/services/session";
 
-var EmberDebug;
-var Ember = window.Ember;
-EmberDebug = Ember.Object.extend({
+const Ember = window.Ember;
+const { Object, run, Application, namespaces } = Ember;
```
Comment (teddyzeenny) [inline ember_debug/main.js]: `Object: EmberObject`

File: ember_debug/main.js [hunk 34]
```diff
@@ -52,27 +52,27 @@ EmberDebug = Ember.Object.extend({
...
+  willDestroy() {
     this.destroyContainer();
     this._super.apply(this, arguments);
   },
 
-  reset: function($keepAdapter) {
+  reset($keepAdapter) {
     this.destroyContainer();
-    Ember.run(this, function() {
+    run(this, function() {
```
Comment (teddyzeenny) [inline ember_debug/main.js]: `run(() => {`

File: ember_debug/models/promise.js [hunk 35]
```diff
@@ -1,13 +1,14 @@
 import computedPolyfill from '../addons/ember-new-computed/index';
-var Ember = window.Ember;
+const Ember = window.Ember;
+const { typeOf, Object, computed, A } = Ember;
```
Comment (teddyzeenny) [inline ember_debug/models/promise.js]: `Object: EmberObject`

File: ember_debug/object-inspector.js [hunk 36]
```diff
@@ -60,9 +51,9 @@ function inspect(value) {
     // `Ember.inspect` is able to handle this use case,
     // but it is very slow as it loops over all props,
     // so summarize to just first 2 props
-    var ret = [], v, count = 0, broken = false;
-    for (var key in value) {
-      if (!('hasOwnProperty' in value) || value.hasOwnProperty(key)) {
+    let ret = [], v, count = 0, broken = false;
```
Comment (teddyzeenny) [inline ember_debug/object-inspector.js]:
```js
let v;
let ret = [];
let count = 0;
let broken = false;
```

File: ember_debug/object-inspector.js [hunk 37]
```diff
@@ -85,7 +76,7 @@ function inspect(value) {
   }
 }
 
-var ObjectInspector = EmberObject.extend(PortMixin, {
+let ObjectInspector = EmberObject.extend(PortMixin, {
```
Comment (teddyzeenny) [inline ember_debug/object-inspector.js]: `export default EmberObject.extend(PortMixin, {`

File: ember_debug/object-inspector.js [hunk 38]
```diff
@@ -247,10 +237,9 @@ var ObjectInspector = EmberObject.extend(PortMixin, {
   },
 
 
-  retainObject: function(object) {
-    var meta = emberMeta(object),
-        guid = guidFor(object),
-        self = this;
+  retainObject(object) {
+    let meta = emberMeta(object),
+        guid = guidFor(object);
```
Comment (teddyzeenny) [inline ember_debug/object-inspector.js]:
```js
let meta = emberMeta(object),
let guid = guidFor(object);
```

File: ember_debug/object-inspector.js [hunk 39]
```diff
@@ -380,17 +369,17 @@ var ObjectInspector = EmberObject.extend(PortMixin, {
...
+    let objectId = message.objectId,
         property = message.property,
         mixinIndex = message.mixinIndex,
         computed = message.computed,
         self = this;
 
-    var object = this.sentObjects[objectId];
+    let object = this.sentObjects[objectId];
 
     function handler() {
```
Comment (teddyzeenny) [inline ember_debug/object-inspector.js]: I think we can remove `self` if we use:
`const handler = () =>`

File: ember_debug/object-inspector.js [hunk 40]
```diff
@@ -475,18 +463,18 @@ function replaceProperty(properties, name, value, options) {
 
   if (found) { properties.splice(i, 1); }
 
-  var prop = { name: name, value: inspectValue(value) };
+  let prop = { name: name, value: inspectValue(value) };
```
Comment (teddyzeenny) [inline ember_debug/object-inspector.js]: `let prop = { name, value: inspectValue(value) };`

File: ember_debug/object-inspector.js [hunk 41]
```diff
@@ -707,7 +695,7 @@ function isComputed(value) {
 }
 
 function toArray(errors) {
-  return keys(errors).map(function(key) {
+  return keys(errors).map(key => {
     return errors[key];
   });
```
Comment (teddyzeenny) [inline ember_debug/object-inspector.js]: This can be:
```js
return keys(errors).map(key => errors[key])
```

File: ember_debug/port.js [hunk 42]
```diff
@@ -1,34 +1,32 @@
-var Ember = window.Ember;
-var oneWay = Ember.computed.oneWay;
-var guidFor = Ember.guidFor;
-var run = Ember.run;
+const Ember = window.Ember;
+const { Object, computed, guidFor, run } = Ember;
```
Comment (teddyzeenny) [inline ember_debug/port.js]: `Object: EmberObject`

File: ember_debug/render-debug.js [hunk 43]
```diff
@@ -1,17 +1,14 @@
 import PortMixin from "ember-debug/mixins/port-mixin";
 import ProfileManager from "ember-debug/models/profile-manager";
 
-var Ember = window.Ember;
-var computed = Ember.computed;
-var oneWay = computed.oneWay;
-var later = Ember.run.later;
+const Ember = window.Ember;
+const { computed: { oneWay }, run: { later }, subscribe, Object } = Ember;
```
Comment (teddyzeenny) [inline ember_debug/render-debug.js]: `Object: EmberObject`

File: ember_debug/render-debug.js [hunk 44]
```diff
@@ -1,17 +1,14 @@
...
 
-var profileManager = new ProfileManager();
-
-var queue = [];
+let profileManager = new ProfileManager(),
+    queue = [];
```
Comment (teddyzeenny) [inline ember_debug/render-debug.js]:
```js
let profileManager = new ProfileManager(),
let queue = [];
```

File: ember_debug/render-debug.js [hunk 45]
```diff
@@ -67,18 +64,17 @@ export default Ember.Object.extend(PortMixin, {
...
-  init: function() {
-    var self = this;
+  init() {
     this._super();
-    this.profileManager.wrapForErrors = function(context, callback) {
-      return self.get('port').wrap(function() {
+    this.profileManager.wrapForErrors = (context, callback) => {
+      return this.get('port').wrap(function() {
         return callback.call(context);
       });
```
Comment (teddyzeenny) [inline ember_debug/render-debug.js]: `return this.get('port').wrap(() => callback.call(context))`

File: ember_debug/route-debug.js [hunk 46]
```diff
@@ -1,27 +1,23 @@
...
-var classify = Ember.String.classify;
-var dasherize = Ember.String.dasherize;
-var computed = Ember.computed;
-var oneWay = computed.oneWay;
-var observer = Ember.observer;
-var later = Ember.run.later;
-
-export default Ember.Object.extend(PortMixin, {
+const Ember = window.Ember;
+const { String: { classify, dasherize }, computed, observer, run: { later }, Object } = Ember;
```
Comment (teddyzeenny) [inline ember_debug/route-debug.js]: `Object: EmberObject`

File: ember_debug/route-debug.js [hunk 47]
```diff
@@ -30,50 +26,49 @@ export default Ember.Object.extend(PortMixin, {
...
       this.sendCurrentRoute();
     }
   },
 
   sendCurrentRoute: observer(function() {
-    var self = this;
-    later(function() {
-      self.sendMessage('currentRoute', { name: self.get('currentPath') });
+    later(() => {
+      this.sendMessage('currentRoute', { name: self.get('currentPath') });
```
Comment (teddyzeenny) [inline ember_debug/route-debug.js]: `this.get('currentPath')`

File: ember_debug/route-debug.js [hunk 48]
```diff
@@ -30,50 +26,49 @@ export default Ember.Object.extend(PortMixin, {
...
+    getTree() {
       this.sendTree();
     },
-    getCurrentRoute: function() {
+    getCurrentRoute() {
       this.sendCurrentRoute();
     }
   },
 
   sendCurrentRoute: observer(function() {
```
Comment (teddyzeenny) [inline ember_debug/route-debug.js]: `observer('currentPath', function() {`

File: ember_debug/services/session.js [hunk 49]
```diff
@@ -1,7 +1,7 @@
-var Ember = window.Ember;
-var EmberObject = Ember.Object;
+const Ember = window.Ember;
+const { Object } = Ember;
```
Comment (teddyzeenny) [inline ember_debug/services/session.js]: `Object: EmberObject`

File: ember_debug/view-debug.js [hunk 50]
```diff
@@ -1,18 +1,16 @@
 /* eslint no-cond-assign:0 */
 import PortMixin from "ember-debug/mixins/port-mixin";
 
-var Ember = window.Ember;
-var guidFor = Ember.guidFor;
+const Ember = window.Ember;
+const { guidFor, $, computed, run, Object, View, typeOf, Component, ViewUtils, A, keys } = Ember;
```
Comment (teddyzeenny) [inline ember_debug/view-debug.js]: `Object: EmberObject`

File: ember_debug/view-debug.js [hunk 51]
```diff
@@ -1,18 +1,16 @@
...
     previewDiv,
-    highlightedElement,
-    $ = Ember.$,
-    later = Ember.run.later,
-    computed = Ember.computed,
-    oneWay = computed.oneWay;
+    highlightedElement;
 
-var ViewDebug = Ember.Object.extend(PortMixin, {
+const ViewDebug = Object.extend(PortMixin, {
```
Comment (teddyzeenny) [inline ember_debug/view-debug.js]: `export default`

File: ember_debug/view-debug.js [hunk 52]
```diff
@@ -122,57 +119,57 @@ var ViewDebug = Ember.Object.extend(PortMixin, {
...
-      viewElem = self.findNearestView(originalTarget);
+    $('body').on('mousemove.inspect-' + this.get('eventNamespace'), e => {
+      let originalTarget = $(e.target);
+      viewElem = this.findNearestView(originalTarget);
       if (viewElem) {
-        self.highlightView(viewElem, true);
+        this.highlightView(viewElem, true);
       }
     })
     .on('mousedown.inspect-' + this.get('eventNamespace'), function() {
```
Comment (teddyzeenny) [inline ember_debug/view-debug.js]: `, () => {`

File: ember_debug/view-debug.js [hunk 53]
```diff
@@ -192,7 +189,7 @@ var ViewDebug = Ember.Object.extend(PortMixin, {
     function pinView() {
```
Comment (teddyzeenny) [inline ember_debug/view-debug.js]: We can remove `self`:
`const pinView = () =>`

File: ember_debug/view-debug.js [hunk 54]
```diff
@@ -297,27 +291,28 @@ var ViewDebug = Ember.Object.extend(PortMixin, {
   },
 
 
-  inspectView: function(view, retained) {
-    var templateName = view.get('templateName') || view.get('_debugTemplateName'),
-        viewClass = shortViewName(view), name;
+  inspectView(view, retained) {
+    const templateName = view.get('templateName') || view.get('_debugTemplateName'),
+          viewClass = shortViewName(view);
```
Comment (teddyzeenny) [inline ember_debug/view-debug.js]:
```js
let templateName = view.get('templateName') || view.get('_debugTemplateName');
let viewClass = shortViewName(view);
```

File: ember_debug/view-debug.js [hunk 55]
```diff
@@ -326,19 +321,19 @@ var ViewDebug = Ember.Object.extend(PortMixin, {
...
         name: shortControllerName(controller),
         completeName: controllerName(controller),
         objectId: this.retainObject(controller)
       };
 
-      var model = this.modelForView(view);
+      let model = this.modelForView(view);
       if (model) {
-        if (Ember.Object.detectInstance(model) || Ember.typeOf(model) === 'array') {
+        if (Object.detectInstance(model) || typeOf(model) === 'array') {
```
Comment (teddyzeenny) [inline ember_debug/view-debug.js]: `EmberObject`

File: ember_debug/view-debug.js [hunk 56]
```diff
@@ -464,10 +458,9 @@ var ViewDebug = Ember.Object.extend(PortMixin, {
   },
 
   // TODO: This method needs a serious refactor/cleanup
-  _highlightRange: function(rect, options) {
-    var div;
-    var isPreview = options.isPreview;
-    var self = this;
+  _highlightRange(rect, options) {
+    let div,
+        isPreview = options.isPreview;
```
Comment (teddyzeenny) [inline ember_debug/view-debug.js]:
```js
let div;
let isPreview = options.isPreview;
```

**Comment by teddyzeenny** [comment]
@Mawaheb this is great!

> I couldn't refactor the contents of /vendor, when I tried, I got an error "Can't use block-scoped let, const out of restricted mode."

Yeah vendor files are not transpiled. We can't (and shouldn't) refactor them.

> Because arrow functions should always be anonymous, I couldn't/figure out how to refactor this one to use this instead of self.

You can declare them as variables instead of functions:
```js
const pinView = () => {
};
Ember.deprecate = (message, test, options) => {
};
```

**Comment by Mawaheb** [comment]
Thank you for your feedback, I modified the code accordingly.

One thing, When I pulled the latest master, and when I rebased, there were some merge/rebase conflicts, I hope I fixed them the right way.

**Comment by teddyzeenny** [comment]
Sorry @Mawaheb this needs another rebase.

**Comment by Mawaheb** [comment]
Sorry for the inconvenience, I hope I got it right this time @teddyzeenny  :)

File: ember_debug/view-debug.js [hunk 57]
```diff
@@ -122,59 +119,59 @@ var ViewDebug = Ember.Object.extend(PortMixin, {
...
+  }),
 
-  willDestroy: function() {
+  willDestroy() {
     this._super();
     $(window).off(this.get('eventNamespace'));
     $(layerDiv).remove();
     $(previewDiv).remove();
     this.get('_lastNodes').clear();
+    View.removeMutationListener(this.viewTreeChanged);
```
Comment (teddyzeenny) [inline ember_debug/view-debug.js]: This should be removed.

File: ember_debug/view-debug.js [hunk 58]
```diff
@@ -226,57 +223,56 @@ var ViewDebug = Ember.Object.extend(PortMixin, {
...
-    this.viewTreeChanged = function() {
-      self.sendTree();
-      self.hideLayer();
+  viewListener() {
+    this.viewTreeChanged = () => {
+      this.sendTree();
+      this.hideLayer();
     };
+
+    View.addMutationListener(this.viewTreeChanged);
```
Comment (teddyzeenny) [inline ember_debug/view-debug.js]: This line should be removed.

File: ember_debug/view-debug.js [hunk 59]
```diff
@@ -226,57 +223,56 @@ var ViewDebug = Ember.Object.extend(PortMixin, {
...
+    let tree;
+    let emberApp = this.get('application');
     if (!emberApp) {
       return false;
     }
 
-    var applicationViewId = $(emberApp.rootElement).find('> .ember-view').attr('id');
-    var rootView = this.get('viewRegistry')[applicationViewId];
+    const applicationViewId = $(emberApp.rootElement).find('> .ember-view').attr('id');
+    let rootView = View.views[applicationViewId];
```
Comment (teddyzeenny) [inline ember_debug/view-debug.js]: Replace with `let rootView = this.get('viewRegistry')[applicationViewId];`

File: ember_debug/view-debug.js [hunk 60]
```diff
@@ -399,13 +395,13 @@ var ViewDebug = Ember.Object.extend(PortMixin, {
     if (!element) { return; }
 
 
-    if (element instanceof Ember.View) {
+    if (element instanceof View) {
       view = element;
     } else {
-      view = this.get('viewRegistry')[element.id];
+      view = View.views[element.id];
```
Comment (teddyzeenny) [inline ember_debug/view-debug.js]: `view = this.get('viewRegistry')[element.id];`

File: ember_debug/view-debug.js [hunk 61]
```diff
@@ -603,39 +598,34 @@ var ViewDebug = Ember.Object.extend(PortMixin, {
...
   /**
    * @method isGlimmer
    * @return {Boolean}
    */
-  _isGlimmer: function() {
-    var id = Ember.keys(this.get('viewRegistry'))[0];
-    return id && !this.get('viewRegistry')[id].get('_childViews');
+  _isGlimmer() {
+    let id = keys(View.views)[0];
+    return id && !View.views[id].get('_childViews');
```
Comment (teddyzeenny) [inline ember_debug/view-debug.js]:
```js
let id = Ember.keys(this.get('viewRegistry'))[0];
return id && !this.get('viewRegistry')[id].get('_childViews');
```

File: ember_debug/view-debug.js [hunk 62]
```diff
@@ -603,39 +598,34 @@ var ViewDebug = Ember.Object.extend(PortMixin, {
...
   },
 
-  viewRegistry: computed('application', function() {
-    return this.get('application.__container__').lookup('-view-registry:main') || Ember.View.views;
-  }),
```
Comment (teddyzeenny) [inline ember_debug/view-debug.js]: the above CP (`viewRegistry`) should not be deleted

**Comment by Mawaheb** [comment]
I made the changes you mentioned, Please let me know if there are any further changes that should be made.

**Comment by teddyzeenny** [comment]
Thanks @Mawaheb!

**Comment by Mawaheb** [comment]
my pleasure :) :+1:

**Pull request merged**